src/components/CraftsList/CraftsList.tsx

```diff
@@ -15,27 +16,52 @@ export function CraftsList() {
     let [accountInfo, setAccountInfo] = useState<AccountInfo>();
     let [profiles, setProfiles] = useState<SkyblockProfile[]>();
     let [selectedProfile, setSelectedProfile] = useState<SkyblockProfile>();
+    let [isLoadingProfileData, setIsLoadingProfileData] = useState(true);
+    let [isLoadingCrafts, setIsLoadingCrafts] = useState(true);
+    let [hasPremium, setHasPremium] = useState(false);
+    let [isLoggedIn, setIsLoggedIn] = useState(false);
 
     useEffect(() => {
+        setIsLoadingCrafts(true);
+        setIsLoadingProfileData(true);
         loadCrafts();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
     function loadCrafts(playerId?: string, profileId?: string) {
         api.getProfitableCrafts(playerId, profileId).then(crafts => {
```

Comment:
crafts are not cached via the profile id thus leading to data unexpectedly being not refreshed when a different profile is selectd